app/gui/src/presenter/graph/state.rs

```diff
@@ -528,20 +528,22 @@ impl<'a> ControllerChange<'a> {
     ) -> Option<node_view::error::Error> {
         use node_view::error::Kind;
         use ExpressionUpdatePayload::*;
-        let (kind, message, trace) = match payload {
-            None | Some(Value) => None,
-            Some(DataflowError { trace }) => Some((Kind::Dataflow, None, trace)),
-            Some(Panic { message, trace }) => Some((Kind::Panic, Some(message), trace)),
-            Some(Pending { .. }) => None,
-        }?;
-        let propagated = if kind == Kind::Panic {
+        let is_propagated = |trace: Vec<AstNodeId>| {
             let nodes = self.nodes.borrow();
             let root_cause = trace.iter().find(|id| nodes.get(**id).is_some());
             !root_cause.contains(&&node_id)
-        } else {
-            // TODO[ao]: traces are not available for Dataflow errors.
-            false
         };
+        let (kind, message, propagated) = match payload {
+            Some(Value { warnings: Some(warnings) }) if warnings.count > 0 =>
+                Some((Kind::Warning, warnings.value, false)),
```

Comment:
The message overrides the visualization value (at least to the docs in the Error structure). But we don't want warnings to override the visualization value, as it is still computable. Therefore, we should return `None` instead of `warnings.value` and add explanation why we're doing so.

Reply:
Fixed the implementation, now warnings are not displayed in error visualization and therefore don't block the normal visualization of a computed value. Video in description updated.